app/models/users/permission_checks.rb

```diff
@@ -86,13 +86,18 @@ def member_of?(project)
   end
 
   def all_permissions_for(context)
-    Authorization
+    permissions = Authorization
       .roles(self, context)
       .includes(:role_permissions)
       .pluck(:permission)
       .compact
-      .uniq
       .map(&:to_sym)
+
+    if context.is_a?(Project) || context.nil?
+      permissions + OpenProject::AccessControl.public_permissions.map(&:name)
+    else
+      permissions
+    end.uniq
```

Comment:
There is one, potentially two, problems with this:
1) Even if the user has no permission in the project at all (e.g. not a member), will the public permissions be added.
2) There is no check on whether the modules the permission belongs to is active in the project. This might not be an actual problem if all callers check for this. I assume that they do as the problem is not specific to the now added public permissions. But please double check this.

Reply:
> Even if the user has no permission in the project at all (e.g. not a member), will the public permissions be added.

Isn't that the case right now as well? When the user is not a member of the project, the non-member permission is returned and the public permissions are also added?